src/chatd.cpp

```diff
@@ -5229,8 +5340,7 @@ void Chat::onUserLeave(Id userid)
 
 void Chat::onAddReaction(Id msgId, Id userId, std::string reaction)
 {
-    Idx messageIdx = msgIndexFromId(msgId);
-    if (messageIdx == CHATD_IDX_INVALID)
+    if (mDbInterface->getIdxOfMsgidFromHistory(msgId) == CHATD_IDX_INVALID)
```

Comment:
Searching for the `msgId` in the DB should be the last option. It's slower than memory, and most of the times the reaction corresponds to a message in memory.

Reply:
Solved in e71ebaf112242c882497ef3fd95b2d7d9b99ef49